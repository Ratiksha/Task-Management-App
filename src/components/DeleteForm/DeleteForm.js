import React, { Component } from 'react';
import './DeleteForm.scss';

export default class DeleteForm extends Component{
    constructor(props){
        super(props);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleSubmit = () => {
        const { listId, taskId, DeleteList, DeleteTask, hideModal } = this.props
        if(DeleteList){
            DeleteList(listId);
        }
        else{
            DeleteTask(listId, taskId)
        }
        hideModal();
    }
    
    render() {
        const { name, DeleteList } = this.props;
        return (
            <form onSubmit={this.handleSubmit}>
                <h3>Delete {DeleteList ? 'List' : 'Task'}</h3>
                <h4>Are you sure you want to delete <span>{name}</span> ?</h4>
                <input type="submit" value="Yes" className="button green delete"/>
            </form>
        )
    }
}