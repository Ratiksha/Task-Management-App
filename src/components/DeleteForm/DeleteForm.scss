h4 span{
    color: var(--color);
}