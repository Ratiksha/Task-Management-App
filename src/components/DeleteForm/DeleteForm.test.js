import React from 'react';
import { shallow } from 'enzyme';
import DeleteForm from './DeleteForm';

let wrapped = shallow(<DeleteForm />);
describe('DeleteForm', () => {
  it('should render the DeleteForm Component correctly', () => {   
    expect(wrapped).toMatchSnapshot();
  });

});