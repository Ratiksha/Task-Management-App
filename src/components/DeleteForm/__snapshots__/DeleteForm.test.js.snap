// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteForm should render the DeleteForm Component correctly 1`] = `ShallowWrapper {}`;
